docs/src/tutorial.rst

```diff
@@ -0,0 +1,496 @@
+          restApiName: 'Widget Service',
+          description: 'This service serves widgets.'
+        });
+
+        // Pass the request to the handler
+        const getWidgetsIntegration = new apigateway.LambdaIntegration(handler);
+
+        // Use the getWidgetsIntegration when there is a GET request
+        api.root.addMethod('GET', getWidgetsIntegration);   // GET /
+      }
+    }
+
+Save it and make sure it builds and creates a (still empty) stack.
+
+.. code-block:: sh
+
+    npm run build
+    cdk synth
+
+.. _add_service:
+
+Step 4: Add the Service to the App
+==================================
+
+To add the service to our app,
+add the following line of code after the existing **import** statement in
+*my_widget_service.ts*.
+
+.. code-block:: ts
+
+    import widget_service = require('../lib/widget_service');
+
+Add the following line of code at the end of the constructor in *my_widget_service.ts*.
+
+.. code-block:: ts
+
+    new widget_service.WidgetService(this, 'Widgets');
+
+Make sure it builds and creates a stack
+(we don't show the stack as it's almost 300 lines).
+
+.. code-block:: sh
+
+    npm run build
+    cdk synth
+
+.. _deploy_and_test:
+
+Step 5: Deploy and Test the App
+===============================
+
+Before you can deploy your first |cdk| app,
+you must bootstrap your deployment,
+which creates some AWS infracture that the |cdk|
+needs.
+See the **bootstrap** section of the :doc:`tools` topic for details.
+
+.. code-block:: sh
+
+    cdk bootstrap
+
+Run the following command to deploy your app.
+
+.. code-block:: sh
+
+    cdk deploy
+
+If the deployment is successfull,
+save the URL for your server, which appears in the last line in the window,
+where GUID is an alpha-numeric GUID and REGION is your region.
+
+.. code-block:: sh
+
+    https://GUID.execute-REGION.amazonaws.com/prod/
+
+You can test your app by getting the list of widgets (currently empty) by navigating to this URL in a
+browser or use the following **curl** command.
+
+.. code-block:: sh
+
+    curl -X GET 'https://GUID.execute-REGION.amazonaws.com/prod'
+
+You can also open the AWS Console,
+navigate to the API Gateway service,
+find **Widget Service** in the list.
+Select **GET** and **Test** to test the function.
+Since we haven't stored any widgets yet, the output should be similar to the following
+(there may be some slight differences in whitespace and quotation marks).
+
+.. code-block:: sh
+
+    { "widgets": [] }
+
+.. _adding_functions:
+
+Step 6: Add the Individual Widget Functions
+===========================================
+
+The next step is to create |LAM| functions to create, show, and delete
+individual widgets.
+Replace the existing **exports.main** function in *widgets.js* with the following code.
+
+.. code-block:: js
+
+    exports.main = function(event, context, callback) {
+      if (event.httpMethod != null) {
+        if (event.httpMethod != "") {
+          if (event.httpMethod == "GET") {
```

Comment:
These three lines below each other make me restless. There should be a better way to express this.

Reply:
I've refactored the code to use a try block.

Reply:
I don't see how that helps.

What's making me nervous here, and I should have been more clear, is you're doing the following:

```
if (variable unequal to one thing)
AND
if (variable unequal to other thing)
AND
if (variable IS EUQAL to something else)
THEN do something
```

Well first of all, if `event.httpMethod == "GET"`, then of course it must be `!= null` and `!= ""`, so there's no need to check those. You could have gotten away with writing:

```ts
if (event.httpMethod == "GET") {
   // ...
}
if (event.httpMethod == "POST") {
  // ...
}
// etc
```
And have the same effect.

The only reason you would type it this way, is because maybe at the end you might write an `else`, something like this:

```ts
if (event.httpMethod != "") {
  if (Event.httpMethod == "GET") {
   // lots of
  }
  // ...lines here...
} else {
  // Do something special
}
```

However, because the inner block is SO LONG, I have a hard time visually scanning whether that's true or not. So I recommend you invert the condition into an early abort, if that's the case:

```ts
if (event.httpMethod == "") {
  // Do something special
  return;
}
if (event.httpMethod == "GET") {
   /// ...
}
if (event.httpMethod == "POST") {
  // ...
}
// etc
```

Long blocks and redundant ifs make me nervous because they might be hiding logic that's easy to miss, and so I recommend that you change the code layout to be very clear and straightforward about what's happening.

I'm not sure how adding a `try` helps there. But also, I can't find the code back in the PR after it has exploded through the merge, so I'll have a look again after it has normalized.

-----

## Also

Forgot to take note of this first time around, but since you're doing JavaScript, REPLACE ALL INSTANCES of the following:

* `==` => `===`
* `!=` => `!==`

The 2-character variants are not to be trusted.